### PR TITLE
Pulses with mirror pattern

### PR DESCRIPTION
**Pull request by chellings**
This PR adds the possibility to specify a mirror_pattern in a pulse, which defines whether and how to mirror the amplitude of the pulse when the pulse is used repeatedly. This can be used to implement toggling unipolar CZ gates.

Inviting @nathlacroix to review.
FYI @stephlazar @antsr

### CODE REVIEW

File: pycqed/measurement/waveform_control/segment.py [hunk 1]
```diff
@@ -727,6 +735,39 @@ def _test_trigger_awg(self):
                 raise ValueError(
                     'There is more than one element on {}'.format(awg))
 
+    def resolve_mirror(self):
+        """
+        Resolves amplitude mirroring for pulses that have a mirror_pattern
+        property.
+        """
```
Comment (nathlacroix) [inline pycqed/measurement/waveform_control/segment.py]: Please detail here and/or where the user would typically add the mirror pattern (in the pulse object, I guess) what patterns are accepted and what they mean. I couldn't find such documentation

**Review by nathlacroix — CHANGES_REQUESTED** [review]
Very useful feature, basic functionality successfully tested on real setup. 

requested change:
- please add documentation (at the most convenient/logical place for user) about the different accepted patterns and what they mean

possible improvements in the future (beyond this pull request):
- I believe the mirror pattern isn't always added as a pulse parameter in the operations dict, meaning that it cannot be easily accessed. Is that correct? It would make sense in the future, to have it for any type of pulse and have it easily accessible.

File: pycqed/measurement/waveform_control/segment.py (hunk 1 above)
Comment (chellings) [reply pycqed/measurement/waveform_control/segment.py]: done

**Review by nathlacroix — COMMENTED** [review]
thanks 👍

**Pull request merged**